### PR TITLE
Add avoid_single_child_in_multi_child_widgets lint rule

### PR DESCRIPTION
**Pull request by shilangyu**

### CODE REVIEW

File: packages/leancode_lint/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 8.1.0
+
+- Enable custom lint `avoid_single_child_in_multi_child_widgets` for `Column`, `Row`, `Flex`, and `Wrap`
```
Comment (shilangyu) [inline packages/leancode_lint/CHANGELOG.md:3]: Feel free to suggest other ones, I can easily add them

Comment (PiotrRogulski) [reply packages/leancode_lint/CHANGELOG.md:3]: Maybe also `Stack`? Usually could be replaced by `Align` or stuff

File: packages/leancode_lint/test/lints_test_app/lib/avoid_single_child_in_multi_child_widget_test.dart [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    );
+  }
+}
+
+class ZeroChildrenIsNotMarked extends StatelessWidget {
+  const ZeroChildrenIsNotMarked({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const Wrap();
```
Comment (PiotrRogulski) [inline packages/leancode_lint/test/lints_test_app/lib/avoid_single_child_in_multi_child_widget_test.dart:67]: Could be a separate lint, but maybe we want to disallow no children as well?

File: packages/leancode_lint/CHANGELOG.md (hunk 1 above)
Comment (PiotrRogulski) [reply packages/leancode_lint/CHANGELOG.md:3]: We could also detect equivalent sliver widgets — `SliverMainAxisGroup` and `SliverCrossAxisGroup`

File: packages/leancode_lint/test/lints_test_app/lib/avoid_single_child_in_multi_child_widget_test.dart (hunk 2 above)
Comment (shilangyu) [reply packages/leancode_lint/test/lints_test_app/lib/avoid_single_child_in_multi_child_widget_test.dart:67]: I thought about it, but figured that there are so many widgets that dont make sense when used with no children/child. Also I would assume it happens way less often, while single child in Column/Row happens quite often by accident. So yeah, if we want a no children/child lint it should be separate

**Pull request merged**